src/components/date-picker/date-picker.tsx

```diff
@@ -486,10 +524,11 @@ export class DatePicker implements LocalizedComponent {
       this.localeData && [
         <calcite-date-picker-month-header
           activeDate={activeDate}
-          headingLevel={this.headingLevel}
-          intlNextMonth={this.intlNextMonth}
-          intlPrevMonth={this.intlPrevMonth}
-          intlYear={this.intlYear}
+          headingLevel={this.headingLevel || HEADING_LEVEL}
+          //t9n props are used here to forward the messages only.
+          intlNextMonth={this.messages.nextMonth}
```

Comment:
Whats the plan for when these are deprecated? Will there be a way to pass these along?

Reply:
Good point . There could be two ways we can do this, one is to keep the `intl` props on `date-picker-month-header` component because it's `internal` or introduce `messageOverrides` and utilize the object to keep it consistent.

Reply:
I think we should introduce messagesOverrides so we get rid of all the intl props

Reply:
this needs a bit of correction. we may introduce `messages` prop in `date-picker-month-header` rather than `messagesOverrides` because the component is internal and user's can only override via the date-picker which already has `messageOverrides` property . `messages` property is the single source of truth and hence it will be parsed down to the child components which are non-localized / non-t9n.

Reply:
Yeah that makes the most sense. These individual small components aren't documented for use by themselves so they can just take the large messages prop from their parent.